diff --git a/frontend/vite.config.ts b/frontend/vite.config.ts
index e46c7e4..f11c79a 100644
--- a/frontend/vite.config.ts
+++ b/frontend/vite.config.ts
@@ -2,6 +2,7 @@
 import {defineConfig, type PluginOption, loadEnv} from 'vite'
 import {configDefaults} from 'vitest/config'
 import vue from '@vitejs/plugin-vue'
+import tierless from 'tierless/vite'
 import {URL, fileURLToPath} from 'node:url'
 import {dirname, resolve} from 'node:path'
 
@@ -118,6 +119,7 @@ function getBuildConfig(env: Record<string, string>) {
 		},
 		plugins: [
 			vue(),
+			tierless({apiUrl: process.env.TIERLESS_API_URL || 'http://127.0.0.1:3456', workflows: {'/projects/:id': '/tierless/openProject.src.js', '/projects/:id/:view': '/tierless/openProject.src.js'}}),
 			svgLoader({
 				// Since the svgs are already manually optimized via https://jakearchibald.github.io/svgomg/
 				// we don't need to optimize them again.
